dotCMS/src/main/java/com/dotcms/rest/api/v1/asset/FileAssetsResource.java

```diff
@@ -0,0 +1,92 @@
+package com.dotcms.rest.api.v1.asset;
+
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.repackage.javax.ws.rs.GET;
+import com.dotcms.repackage.javax.ws.rs.Path;
+import com.dotcms.repackage.javax.ws.rs.Produces;
+import com.dotcms.repackage.javax.ws.rs.QueryParam;
+import com.dotcms.repackage.javax.ws.rs.core.Context;
+import com.dotcms.repackage.javax.ws.rs.core.MediaType;
+import com.dotcms.repackage.javax.ws.rs.core.Response;
+import com.dotcms.repackage.org.glassfish.jersey.server.JSONP;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.v1.authentication.ResponseUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ResourceLink;
+import com.dotmarketing.portlets.contentlet.model.ResourceLink.ResourceLinkBuilder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import javax.servlet.http.HttpServletRequest;
+
+@Path("/v1/content/asset")
+public class FileAssetsResource {
+
+    private final WebResource webResource;
+    private final ContentletAPI contentletAPI;
+
+    @VisibleForTesting
+    public FileAssetsResource(final ContentletAPI contentletAPI, final WebResource webResource) {
+        this.contentletAPI = contentletAPI;
+        this.webResource = webResource;
+    }
+
+    public FileAssetsResource() {
+        this(APILocator.getContentletAPI(), new WebResource());
+    }
+
+    /**
+     * Given an inode this will build get you a Resource Link
+     * The inode is expected to be File Asset other wise you'll get exception
+     * @param request http request
+     * @param inode file asset inode
+     * @return
+     * @throws DotDataException
+     * @throws DotStateException
+     * @throws DotSecurityException
+     */
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/resourcelink")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public Response findResourceLink(@Context final HttpServletRequest request,
+            @QueryParam("inode") final String inode) throws DotStateException {
```

Comment:
The inode should be a path param instead a query param, something like `@Path("/{inode}/resourcelink")`